osu.Game/Overlays/Settings/SettingsNumberBox.cs

```diff
@@ -67,7 +67,7 @@ public NumberControl()
 
         private class OutlinedNumberBox : OutlinedTextBox
         {
-            protected override bool CanAddCharacter(char character) => char.IsNumber(character);
+            protected override bool CanAddCharacter(char character) => character >= '0' && character <= '9';
```

Comment:
Should probably have an inline comment attached explaining why `char.IsNumber` is not used here, as to not fall into this in the future.

Reply:
Also if that's the case, it should probably be in some kind of helper method with documentation as to why we check this way.